### PR TITLE
UCP/WIREUP/SOCKADDR: refactor ucp_conn_request_t

### PR DESCRIPTION
**Pull request by evgeny-leksikov**
## What
Store tl_iface instead of wiface_idx lookup on connect request callback

## Why ?
Simplify code, especially for https://github.com/openucx/ucx/pull/3946

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/11697/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>PASSED</b#  > on 29 workers (click for details)</summary>

**Note:** the logs will be deleted after 09-Aug-2019

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/b7027933261624ea47c2a84af510ee5b/raw/0466dd00521fb2573da7bd1b057e2a13d6f825e1/_main) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/b7027933261624ea47c2a84af510ee5b/raw/0a6992a52fac9d3eb9d6c39aa5e96e41a0b1b9fb/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/b7027933261624ea47c2a84af510ee5b/raw/e581dee8ee13eb711cc39390717a88dfd9d86c16/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/b7027933261624ea47c2a84af510ee5b/raw/7616676b6edbde580f87598f7592bc1c791c6377/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/b7027933261624ea47c2a84af510ee5b/raw/a66cd4641f2b545342d1c2bb2ada571c468852d2/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/b7027933261624ea47c2a84af510ee5b/raw/731d80049b7a047d29b84f62c86a3a80d30757de/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/b7027933261624ea47c2a84af510ee5b/raw/62fe79a259b5182cfd554f629849bfba711b6188/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/b7027933261624ea47c2a84af510ee5b/raw/f1dedec65ef3f164a34263fa5bacf52750b0f4dc/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/b7027933261624ea47c2a84af510ee5b/raw/b081338240affd3bd5d89d53eabd915921aafa3c/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W0](https://gist.githubusercontent.com/mellanox-github/b7027933261624ea47c2a84af510ee5b/raw/5af1cceaa42cf5336f2b5b2f2712e12df6b55bd7/hpc-test-althca_W0) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W1](https://gist.githubusercontent.com/mellanox-github/b7027933261624ea47c2a84af510ee5b/raw/1fabc512e816633025c93f0190a6580b91815a37/hpc-test-althca_W1) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W2](https://gist.githubusercontent.com/mellanox-github/b7027933261624ea47c2a84af510ee5b/raw/794fbdb92c818beb388b710d26601c74322589ed/hpc-test-althca_W2) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W3](https://gist.githubusercontent.com/mellanox-github/b7027933261624ea47c2a84af510ee5b/raw/aaf5d677cf1672214e5fdd7d214792e1b27afba7/hpc-test-althca_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/b7027933261624ea47c2a84af510ee5b/raw/b6a1d7b1a28c2572ac7afa98c208051535abf54e/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/b7027933261624ea47c2a84af510ee5b/raw/1b6b481c84a56b478e0e035568638f5d61dfe503/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/b7027933261624ea47c2a84af510ee5b/raw/c8f8fd804b349c21122a64020b97c324aa2c17d7/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/b7027933261624ea47c2a84af510ee5b/raw/1b5288e5ae20135c5d5f2bcb23e0921aba6ef097/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/b7027933261624ea47c2a84af510ee5b/raw/86dfc5985d6b496c83fbea5359bb6c8df9177830/hpc-test-node-legacy_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/b7027933261624ea47c2a84af510ee5b/raw/a3253b441ea7b4573efd65749b0461b8ee1c45b4/hpc-test-node-legacy_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/b7027933261624ea47c2a84af510ee5b/raw/b4cb5af35aec457a715256ba1f7778139762568c/hpc-test-node-legacy_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/b7027933261624ea47c2a84af510ee5b/raw/83052c8e2c9392e3b553f53d873c3eeaf892269c/hpc-test-node-legacy_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/b7027933261624ea47c2a84af510ee5b/raw/5a3c96b67dfd64dda05b42659ae587513f882e67/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/b7027933261624ea47c2a84af510ee5b/raw/071b83c8aeb754e72dd3efe3ae45f9bdbeaf20c7/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/b7027933261624ea47c2a84af510ee5b/raw/c3603bf820b8b7d68819d3562bd050095eeb061a/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/b7027933261624ea47c2a84af510ee5b/raw/5d38598aedaaef95c98ea05395649fe2f9b12ddd/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/b7027933261624ea47c2a84af510ee5b/raw/5bd4300061942bfd1971d55606ca155364b62184/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/b7027933261624ea47c2a84af510ee5b/raw/36143adcd943ace93d9e4d238bf889948cffe18e/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/b7027933261624ea47c2a84af510ee5b/raw/4291910582f976323348ee0995b64581ef02b89f/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/b7027933261624ea47c2a84af510ee5b/raw/d5456778be75c0c87131b67bf92a180170d23e50/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

File: src/ucp/core/ucp_ep.h [hunk 1]
```diff
@@ -362,10 +362,7 @@ typedef struct ucp_wireup_client_data {
 typedef struct ucp_conn_request {
     ucp_listener_h              listener;
     uct_conn_request_h          uct_req;
-    uint8_t                     wiface_idx;  /**< Index of the listening wiface in
-                                                  the array of listening wifaces,
-                                                  on which the connection request
-                                                  was received on */
+    uct_iface_h                 uct_iface;
```
Comment (alinask) [inline src/ucp/core/ucp_ep.h]: minor and for next PR: please document these fields.

**Comment by alinask** [comment]
👍

**Pull request merged**